spec/spec_helper.rb

```diff
@@ -254,6 +254,24 @@ module Shell
     Chef.resource_priority_map.instance_variable_set(:@map, resource_priority_map.dup)
   end
 
+  # This bit of jankiness guards against specs which accidentally drop privs when running as
+  # root -- which are nearly impossible to debug and so we bail out very hard if this
+  # condition ever happens.  If a spec stubs Process.[e]uid this can throw a false positive
+  # which the spec must work around by unmocking Process.[e]uid to and_call_original in its
+  # after block.
+  if Process.euid == 0 && Process.uid == 0
+    config.after(:each) do
```

Comment:
ISTR these things can take block arguments which are the RSpec objects representing the test. If that's true then you could use that to give a bit more info about the culprit test, which would probably be really awesome if you needed it.

Reply:
i checked the rspec docs and couldn't find any block arguments to these....   i didn't go code surfing to see if they were hidden options...